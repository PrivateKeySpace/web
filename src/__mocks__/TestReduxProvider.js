import React from 'react'
import PropTypes from 'prop-types'
import { createStore } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'
import rootReducer from '../store/rootReducer'

const displayName = '__mocks__/TestReduxProvider'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ state: PropTypes.object, children: PropTypes.any }),
  defaultProps({ state: {} })
)

function TestReduxProvider (props) {
  const { state, children } = props
  const store = createStore(rootReducer, state)

  return (
    <ReduxProvider store={store}>
      {children}
    </ReduxProvider>
  )
}

export default enhance(TestReduxProvider)
