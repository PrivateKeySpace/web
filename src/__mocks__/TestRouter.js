import React from 'react'
import PropTypes from 'prop-types'
import { MemoryRouter } from 'react-router'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'

const displayName = '__mocks__/TestRouter'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ path: PropTypes.string, children: PropTypes.any }),
  defaultProps({ path: '/' })
)

function TestRouter (props) {
  const { path, children } = props

  return (
    <MemoryRouter initialEntries={[path]} initialIndex={0}>
      {children}
    </MemoryRouter>
  )
}

export default enhance(TestRouter)
