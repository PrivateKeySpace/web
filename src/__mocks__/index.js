export { default as TestRouter } from './TestRouter'
export { default as TestReduxProvider } from './TestReduxProvider'
