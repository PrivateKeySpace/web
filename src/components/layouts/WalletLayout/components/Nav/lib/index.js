export { default as mapProps } from './mapProps'
export { default as defaultPropsSpec } from './defaultPropsSpec'
