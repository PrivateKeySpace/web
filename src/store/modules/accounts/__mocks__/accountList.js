const accountList = [
  {
    'path': [
      2147483697,
      2147483648,
      2147483648
    ],
    'address': '3Pc8vh9xUopMmiBW1yDAm3h1aqqbmmv11p',
    'addressPath': [
      2147483697,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6Ca3FEzN1SemuKNujpAeyDLXk7DXXtgpUs5HHLrG5brjVQm7ib3SLqMKfWWQZset7HMm7wM58wVGEJm5eG6rkxEcYvtvExYaM9T7jxkmmMp',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': true
  },
  {
    'path': [
      2147483692,
      2147483648,
      2147483648
    ],
    'address': '1Bswvjfoa1p1kMkwE4C8KAPbYLSKfg1Nni',
    'addressPath': [
      2147483692,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6D4zdJZuWGKQK5qMyj7reDcapRE6StaVTGS3xWCAjRyLfne84tGxUK1yYNrUWFY8gYtJA7xd3s97o1uiPkFEhpdsuDsNkLnj5iHyAbCEnRv',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': false
  }
]

export default accountList
