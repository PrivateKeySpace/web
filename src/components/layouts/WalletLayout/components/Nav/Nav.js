import React from 'react'
import { compose, setDisplayName, mapProps, pure, defaultProps } from 'recompose'
import { withRouter } from 'react-router'
import { mapProps as propsMapper, defaultPropsSpec } from './lib'
import { Item } from './components'

const navItems = [
  { name: 'Dashboard', path: '/wallet/dashboard' },
  { name: 'Personal accounts', path: '/wallet/personal' },
  { name: 'Multisig accounts', path: '/wallet/multisig' },
  { name: 'Cosigners', path: '/wallet/cosigners' },
  { name: 'Clients', path: '/wallet/clients' },
  { name: 'Help', path: '/wallet/help' }
]

const displayName = 'components/layouts/WalletLayout/components/Nav'

const enhance = compose(
  setDisplayName(displayName),
  withRouter,
  defaultProps(defaultPropsSpec),
  mapProps(propsMapper),
  pure
)

function Nav (props) {
  const { activePath } = props

  return <nav>
    {navItems.map(navItem => (
      <Item
        key={navItem.path}
        to={navItem.path}
        isActive={navItem.path === activePath}
      >
        {navItem.name}
      </Item>
    ))}
  </nav>
}

export default enhance(Nav)
