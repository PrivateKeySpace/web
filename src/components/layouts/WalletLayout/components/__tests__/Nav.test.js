import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { TestRouter } from '../../../../../__mocks__'
import Nav from '../Nav'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const nav = render(
    <TestRouter path='/wallet/dashboard'>
      <Nav />
    </TestRouter>
  )
  expect(nav).toMatchSnapshot()
})
