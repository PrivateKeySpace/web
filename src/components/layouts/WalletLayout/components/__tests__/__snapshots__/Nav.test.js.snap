// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<nav>
  <a
    href="/wallet/dashboard"
  >
    <div
      class="p3_vx7tlw-o_O-pl4_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl-o_O-fontWeight500_gt7myn-o_O-active_15dxwys"
    >
      Dashboard
    </div>
  </a>
  <a
    href="/wallet/personal"
  >
    <div
      class="p3_vx7tlw-o_O-pl4_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
    >
      Personal accounts
    </div>
  </a>
  <a
    href="/wallet/multisig"
  >
    <div
      class="p3_vx7tlw-o_O-pl4_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
    >
      Multisig accounts
    </div>
  </a>
  <a
    href="/wallet/cosigners"
  >
    <div
      class="p3_vx7tlw-o_O-pl4_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
    >
      Cosigners
    </div>
  </a>
  <a
    href="/wallet/clients"
  >
    <div
      class="p3_vx7tlw-o_O-pl4_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
    >
      Clients
    </div>
  </a>
  <a
    href="/wallet/help"
  >
    <div
      class="p3_vx7tlw-o_O-pl4_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
    >
      Help
    </div>
  </a>
</nav>
`;
